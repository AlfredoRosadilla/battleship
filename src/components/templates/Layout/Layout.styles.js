import styled from 'styled-components';

export const Container = styled.div`
  align-items: center;
  display: flex;
  flex-direction: column;
  justify-content: flex-start;
  min-height: 100vh;
  width: 100vw;

  a {
    color: ${({ theme }) => theme.colors.primary};
  }
`;
