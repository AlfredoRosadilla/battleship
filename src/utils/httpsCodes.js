export default (function () {
  let codes = {
    SUCCESS: {
      200: 'OK',
      201: 'Created',
      202: 'Accepted',
      204: 'No Content',
    },
    REDIRECTION: {
      301: 'Moved Permanently',
      302: 'Found',
      304: 'Not Modified',
    },
    CLIENT_ERROR: {
      400: 'Bad Request',
      401: 'Unauthorized',
      403: 'Forbidden',
      404: 'Not Found',
      405: 'Method Not Allowed',
      406: 'Not Acceptable',
      409: 'Conflict',
      410: 'Gone',
      429: 'Too Many Requests',
    },
    SERVER_ERROR: {
      500: 'Internal Server Error',
      501: 'Not Implemented',
      502: 'Bad Gateway',
      503: 'Service Unavailable',
      504: 'Gateway Timeout',
    },
  };

  Object.keys(codes).forEach((httpGroup) => {
    Object.keys(codes[httpGroup]).forEach((httpCode) => {
      codes[httpGroup][
        httpCode
      ] = `${httpCode} - ${codes[httpGroup][httpCode]}`;
    });
  });

  const RAW_HTTP_CODES = {
    ...codes.SUCCESS,
    ...codes.REDIRECTION,
    ...codes.CLIENT_ERROR,
    ...codes.SERVER_ERROR,
  };

  return {
    ...codes,
    getAllCodes() {
      return RAW_HTTP_CODES;
    },
    getCodeDescription(code) {
      return this.getAllCodes()[code];
    },
    getCodeByDescription(description) {
      const codeKeys = Object.keys(RAW_HTTP_CODES);

      return codeKeys.find((codeKey) =>
        RAW_HTTP_CODES[codeKey]
          .toLowerCase()
          .endsWith(description.toLowerCase()),
      );
    },
  };
})();
