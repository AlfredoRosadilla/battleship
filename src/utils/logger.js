// Importamos el módulo de sistema de archivos de Node.js
const fs = require('fs');
const path = require('path');

const logsPath = path.join(`${__dirname}/../../`, 'logs');

// Definimos un enumerador para los niveles de log
const LogLevels = {
  INFO: 'info',
  WARNING: 'warning',
  ERROR: 'error',
};

const ConsoleLevels = {
  [LogLevels.INFO]: 'log',
  [LogLevels.ERROR]: 'error',
  [LogLevels.WARNING]: 'warning',
};

class Logger {
  constructor() {
    this.logsFolder = logsPath;

    // Creamos la carpeta de logs si no existe
    if (!fs.existsSync(this.logsFolder)) {
      fs.mkdirSync(this.logsFolder, { recursive: true });
    }
  }

  // Método para agregar un log
  addLog(log, level) {
    // Creamos un mensaje de log con la fecha y hora actuales
    const logMessage = `${new Date().toISOString()} - ${log}\n`;

    // Definimos el nombre del archivo de log
    const logFile = path.join(
      this.logsFolder,
      level,
      `${new Date().toISOString().split('T')[0]}.log`,
    );

    // Creamos la carpeta para el nivel de log si no existe
    const logFolder = path.dirname(logFile);
    if (!fs.existsSync(logFolder)) {
      fs.mkdirSync(logFolder, { recursive: true });
    }

    // Escribimos el mensaje de log en el archivo de log
    fs.appendFileSync(logFile, logMessage);

    // Mostramos el mensaje de log en la consola
    console[ConsoleLevels[level]](logMessage);
  }
}

// Exportamos el Logger y los LogLevels
module.exports = { Logger, LogLevels };
