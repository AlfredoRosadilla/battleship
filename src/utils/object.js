export function removeUndefined(baseObject) {
  return Object.entries(baseObject).reduce((acc, [key, value]) => {
    if (value !== undefined) {
      acc[key] = value;
    }
    return acc;
  }, {});
}
