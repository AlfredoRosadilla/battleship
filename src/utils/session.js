export const COOKIE_SESSION_NAME = 'Inmobiliarias-Golden-Sky-session-token';
export const COOKIE_USER_DATA_NAME =
  'Inmobiliarias-Golden-Sky-session-user-data';
