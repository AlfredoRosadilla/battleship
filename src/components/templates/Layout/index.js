import React from 'react';
import PropTypes from 'prop-types';

import { Container } from './Layout.styles';
import Header from './../../organisms/Header';
import Footer from './../../organisms/Footer';

/**
 * @typedef {Object} props
 * @property {JSX} children element
 * @property {string} testId, id to use as testing reference
 */

/**
 * Layout
 * @param {props} props
 * @returns Layout component instance
 */
const Layout = ({ children, testId, className }) => (
  <Container className={className} data-testid={`${testId}-container`}>
    <Header testId={`${testId}-header`} />
    {children}
    <Footer testId={`${testId}-footer`} />
  </Container>
);

Layout.propTypes = {
  testId: PropTypes.string.isRequired,
  children: PropTypes.element.isRequired,

  className: PropTypes.string,
};

Layout.defaultProps = {
  className: '',
};

export default Layout;
