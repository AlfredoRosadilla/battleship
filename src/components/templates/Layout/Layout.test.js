import React from 'react';

import { render } from './../../../utils/testUtils';

import Layout, { Methods } from './index';

describe('<Layout />', () => {
  let wrapper;
  let defaultProps;
  const getJSONWrapper = () => wrapper.container.firstChild;
  const populateWrapper = (props) => {
    wrapper = render(<Layout {...Object.assign(defaultProps, props)} />);
  };

  beforeEach(() => {
    defaultProps = {
      testId: 'test-id',
      children: <div />,
    };
  });

  describe('Basic', () => {
    it('should have not log errors in console', () => {
      const spy = jest.spyOn(global.console, 'error');

      populateWrapper();
      expect(spy).not.toHaveBeenCalled();
    });

    it('should default export to be a function', () => {
      expect(typeof Layout).toEqual('function');
    });
  });

  describe('DOM elements', () => {
    ['container'].forEach((testIdSufix) => {
      it(`should have an element with testId sufix: ${testIdSufix}`, () => {
        populateWrapper();

        expect(
          wrapper.getByTestId(`${defaultProps.testId}-${testIdSufix}`),
        ).not.toBe(null);
      });
    });
  });

  describe('Snapshot testing', () => {
    it('with defaultProps', () => {
      populateWrapper();

      expect(getJSONWrapper()).toMatchSnapshot();
    });

    it('with className', () => {
      populateWrapper({
        className: 'test-class-name',
      });

      expect(getJSONWrapper()).toMatchSnapshot();
    });
  });
});
