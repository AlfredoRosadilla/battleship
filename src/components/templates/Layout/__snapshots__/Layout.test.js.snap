// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Layout /> Snapshot testing with className 1`] = `
<div
  class="sc-bdvvtL jluOGi test-class-name"
  data-testid="test-id-container"
>
  <div />
</div>
`;

exports[`<Layout /> Snapshot testing with defaultProps 1`] = `
<div
  class="sc-bdvvtL jluOGi"
  data-testid="test-id-container"
>
  <div />
</div>
`;
